### PR TITLE
BUG: -unsigned_int(0) no overflow warning

### PR DESCRIPTION
**Pull request by seberg**
There is no reason `-np.uint8(0)`, etc. should give a warning.

Also removes a compile time warning (on clang at least).  The reason for this was checking which integer operations are missing the warnings.
It seems the only one is now the `**` operator but adding it there should be a separate PR, and needs a bit of thought.

### CODE REVIEW

File: doc/neps/nep-0050-scalar-promotion.rst [hunk 1]
```diff
@@ -98,8 +98,8 @@ more important than the casting change itself.
     * ``with np._no_nep50_warning():`` allows to suppress warnings when
       ``weak_and_warn`` promotion is used.  (Thread and context safe.)
 
-    At this time overflow warnings on some scalar integer operations are still
-    missing.  Further, ``np.can_cast`` fails to give warnings in the
+    At this time overflow warnings on integer power missing.
```
Comment (mattip) [inline doc/neps/nep-0050-scalar-promotion.rst:101]:
```suggestion
    At this time overflow warnings on integer power are missing.
```

**Review by mattip — APPROVED** [review]
LGTM, just one documentation nit. The cygwin run has started failing across the board.

**Comment by mattip** [comment]
Thanks @seberg

**Comment by charris** [comment]
> win32 is failing to run nm.exe

That happens now and then.

**Pull request merged**